### PR TITLE
W-11567192: Refactor general component tracing and span creation infr…

### PR DESCRIPTION
**Pull request by fsgonz**
…astructure

### CODE REVIEW

**Review by IvanAndresFritzler — APPROVED** [review]

**Review by sofiamorseletto — DISMISSED** [review]

**Review by IvanAndresFritzler — DISMISSED** [review]

File: integration/src/test/java/org/mule/test/components/tracing/AsyncErrorTracingTestCase.java [hunk 1]
```diff
@@ -75,6 +76,10 @@ public void testAsyncFlow() throws Exception {
           exportedSpans.stream().filter(span -> span.getName().equals(EXPECTED_ASYNC_SPAN_NAME)).findFirst()
               .orElse(null);
 
+      CapturedExportedSpan asyncSpanRpute =
```
Comment (sofiamorseletto) [inline integration/src/test/java/org/mule/test/components/tracing/AsyncErrorTracingTestCase.java:79]: Small typo

Comment (fsgonz) [reply integration/src/test/java/org/mule/test/components/tracing/AsyncErrorTracingTestCase.java]: Done

**Comment by fsgonz** [comment]
--validate

**Comment by fsgonz** [comment]
--validate

**Review by IvanAndresFritzler — APPROVED** [review]

**Pull request merged**